mne/annotations.py

```diff
@@ -835,28 +835,27 @@ def _update_hed_strings(self, hed_strings):
                 f"Number of HED strings ({len(hed_strings)}) must match the number of "
                 f"annotations ({len(self)})."
             )
-        # validation of HED strings
+        # create HedString objects
         schema = self.hed.load_schema_version(self.hed_version)
+        self._hed_strings = [self.hed.HedString(hs, schema) for hs in hed_strings]
+        # validation of HED strings
         validator = HedValidator(schema)
         error_handler = self.hed.errors.ErrorHandler(check_for_warnings=False)
-        error_strs = [
-            self._validate_one_hed_string(hs, schema, validator, error_handler)
-            for hs in hed_strings
+        issues = [
+            validator.validate(
+                hs, allow_placeholders=False, error_handler=error_handler
+            )
+            for hs in self._hed_strings
         ]
-        if any(map(len, error_strs)):
+        error_strings = [self.hed.get_printable_issue_string(issue) for issue in issues]
```

Comment:
You could just do: 
`error_string = self.hed.get_printable_issue_string(issues)`

Also, do you want to add links to the error descriptions in the HED specification?  If so, you could do:

`error_string = self.hed.get_printable_issue_string(issues, add_link=True)`

Are there setting in MNE for just reporting errors and not warnings?